docker/prepare_proxy.sh

```diff
@@ -45,59 +40,52 @@ while getopts ":p:u:e:i:h" opt; do
 
 # Create a new chain for selectively redirecting outbound packets to
-# istio and attach it the OUTPUT rule for all tcp traffic. '-j RETURN'
-# bypasses the proxy port; '-j ISTIO_REDIRECT' redirects to the proxy
-# port.
-iptables -t nat -N ISTIO_OUTPUT
-iptables -t nat -A OUTPUT -p tcp -j ISTIO_OUTPUT                                    -m comment --comment "istio/install-istio-output"
+# Envoy.
+iptables -t nat -N ISTIO_OUTPUT                                               -m comment --comment "istio/common-output-chain"
+
+# Jump to the ISTIO_OUTPUT chain from OUTPUT chain for all tcp
+# traffic. '-j RETURN' bypasses Envoy and '-j ISTIO_REDIRECT'
+# redirects to Envoy.
+iptables -t nat -A OUTPUT -p tcp -j ISTIO_OUTPUT                              -m comment --comment "istio/install-istio-output"
 
-# Locally routed traffic is not captured by PREROUTING chain. Redirect
-# loopback back traffic when the DST_IP is not explicitly the loopback
-# address to handles appN => proxy (client) => proxy (server) => appN.
-iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_REDIRECT           -m comment --comment "istio/implicit-loopback"
+# Redirect app calls to back itself via Envoy when using the service VIP or endpoint
+# address, e.g. appN => Envoy (client) => Envoy (server) => appN.
+iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_REDIRECT     -m comment --comment "istio/redirect-implicit-loopback"
 
-# Avoid infinite loops. Don't redirect proxy traffic directly back to
-# the proxy for non-loopback traffic.
-iptables -t nat -A ISTIO_OUTPUT -m owner --uid-owner ${ISTIO_PROXY_UID} -j RETURN   -m comment --comment "istio/exclude-proxy"
+# Avoid infinite loops. Don't redirect Envoy traffic directly back to
+# Envoy for non-loopback traffic.
+iptables -t nat -A ISTIO_OUTPUT -m owner --uid-owner ${ENVOY_UID} -j RETURN   -m comment --comment "istio/bypass-envoy"
 
-# Skip redirection for proxy-aware applications and
-# container-to-container traffic which explicitly use localhost.
-iptables -t nat -A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN                           -m comment --comment "istio/explicit-loopback"
+# Skip redirection for Envoy-aware applications and
+# container-to-container traffic both of which explicitly use
+# localhost.
+iptables -t nat -A ISTIO_OUTPUT -d 127.0.0.1/32 -j RETURN                     -m comment --comment "istio/bypass-explicit-loopback"
 
-# The default outobund redirection policy is opt-out. IP ranges may be
-# selectively excluded from redirection with ISTIO_IP_RANGES_EXCLUDE.
-#
-# If ISTIO_IP_RANGES_INCLUDE is defined the outbound redirection
-# policy is default opt-in. IP ranges may be selectively included for
-# redirection with ISTIO_IP_RANGES_INCLUDE.
+# Only redirect outbound traffic to selective IP ranges when
+# ISTIO_IP_RANGES_INCLUDE is non-empty. Otherwise redirect all
+# outbound traffic not already handled by the previous rules to Envoy.
```

Comment:
I would reuse a previous version of this comment as it looked better.
All outbound traffic will be redirected to Envoy by default. If ISTIO_IP_RANGES_INCLUDE is non-empty, only traffic bound for the destinations specified in this list will be captured.